Pcap++/src/PcapFilter.cpp

```diff
@@ -153,93 +154,20 @@ std::string IFilterWithOperator::parseOperator()
 	}
 }
 
-void IPFilter::convertToIPAddressWithMask(std::string& ipAddrmodified, std::string& mask) const
-{
-	if (m_IPv4Mask.empty())
-		return;
-
-	// Handle the mask
-
-	// The following code lines verify both ipAddress and ipv4Mask are valid IPv4 addresses
-	// The IPv4 limitation comes from the fact libPcap/WinPcap/Npcap doesn't support mask for IPv6 addresses
-
-	IPv4Address ipAddr;
-	try
-	{
-		ipAddr = IPv4Address(m_Address);
-	}
-	catch(const std::exception&)
-	{
-		PCPP_LOG_ERROR("Invalid IP address '" << m_Address << "', setting the mask to an empty value");
-		mask.clear();
-		return;
-	}
-
-	IPv4Address maskAsAddr;
-	try
-	{
-		maskAsAddr = IPv4Address(m_IPv4Mask);
-	}
-	catch(const std::exception&)
-	{
-		PCPP_LOG_ERROR("Invalid IPv4 mask '" << m_IPv4Mask << "', setting the mask to an empty value");
-		mask.clear();
-		return;
-	}
-
-	// If all addresses are IPv4 valid addresses, make sure ipAddress matches the mask. If it's not, mask the address with the mask
-	// The reason for doing that is libPcap/WinPcap/Npcap doesn't allow filtering an IP address that doesn't match the mask
-
-	uint32_t addrAsIntAfterMask = ipAddr.toInt() & maskAsAddr.toInt();
-	ipAddrmodified = IPv4Address(addrAsIntAfterMask).toString();
-}
-
-void IPFilter::convertToIPAddressWithLen(std::string& ipAddrmodified) const
-{
-	if (m_Len == 0)
-		return;
-
-	// Handle the length
-
-	// The following code lines verify IP address is valid (IPv4 or IPv6)
-
-	IPAddress ipAddr;
-	try
-	{
-		ipAddr = IPAddress(ipAddrmodified);
-	}
-	catch(const std::exception&)
-	{
-		PCPP_LOG_ERROR("Invalid IP address '" << ipAddrmodified << "', setting len to zero");
-		return;
-	}
-
-	if (ipAddr.getType() == IPAddress::IPv4AddressType)
-	{
-		uint32_t addrAsInt = ipAddr.getIPv4().toInt();
-		uint32_t mask = static_cast<uint32_t>(-1) >> ((sizeof(uint32_t) * 8) - m_Len);
-		addrAsInt &= mask;
-		ipAddrmodified = IPv4Address(addrAsInt).toString();
-	}
-}
-
 void IPFilter::parseToString(std::string& result)
 {
 	std::string dir;
-	std::string ipAddr = m_Address;
-	std::string mask = m_IPv4Mask;
-	convertToIPAddressWithMask(ipAddr, mask);
-	convertToIPAddressWithLen(ipAddr);
+	std::string ipAddr = m_Network.toString();
```

Comment:
That was how it uses to be, but it adds a branch condition as `toString` always generates with the prefix length. Is there a performance difference between filtering just an `net IP` and `net IP/Prefix`?

Reply:
I'm not sure if there is a performance difference, there might be 🤷‍♂️ 
I'm less worried about adding a branch condition to `toString()` because it's usually not called in the critical path, it's just parsing and setting the filter. Using `net IP` is nicer than `net IP/prefix` in my opinion...

Reply:
Well, I am not talking about adding a branch to `toString()` but around it, as it would need to get just the network prefix if the subnet is full size, instead of calling the `IPNetwork`'s `toString()`.
IMO tho, it makes it more consistent and predictable if the filter always generates the format `net IP/Prefix` instead of "generates `net IP/Prefix` unless the prefix is full, then its just `net IP`" ¯\_(ツ)_/¯

Reply:
> Well, I am not talking about adding a branch to `toString()` but around it in the call site, as it would need to get just the network prefix if the subnet is full size, instead of calling the `IPNetwork`'s `toString()`.

I'm not sure I understand this sentence...



> IMO tho, it makes it more consistent and predictable if the filter's `parseToString()` always generates the format `net IP/Prefix` instead of "generates `net IP/Prefix` unless the prefix is full, then its just `net IP`" ¯\_(ツ)_/¯

I don't have a strong opinion here so we can go with your preference 🙂

Reply:
> I'm not sure I understand this sentence...

Eh, it was not that important anyway.

> I don't have a strong opinion here so we can go with your preference 🙂

Alrighty then, so I guess we are keeping it as is currently, unless there is a significant performance impact, which I don't think there should be, but you never know. ¯\\\_(ツ)\_/¯

Mark this as resolved?